DevCheck.ps1

```diff
@@ -186,6 +207,105 @@ function Get-CpuArchitecture
     }
 }
 
+function Test-SemVer
+{
+    param(
+        [string]$left,
+        [string]$right
+    )
+
+    $lefty = $left.Split('.')
+    $righty = $right.Split('.')
+
+    $cmp = [int]$lefty[0] - [int]$righty[0]
+    if ($cmp -ne 0)
+    {
+        return $cmp
+    }
+
+    $cmp = [int]$lefty[1] - [int]$righty[1]
+    if ($cmp -ne 0)
+    {
+        return $cmp
+    }
+
+    $cmp = [int]$lefty[1] - [int]$righty[1]
+    if ($cmp -ne 0)
+    {
+        return $cmp
+    }
+
+    $lefty2 = $lefty -Match "(\d+)(-[^+]+)?(\+.+)?$"
+    $righty2 = $righty -Match "(\d+)(-[^+]+)?(\+.+)?$"
+    $cmp = [int]$lefty2[0] - [int]$righty2[0]
+    if ($cmp -ne 0)
+    {
+        return $cmp
+    }
+    if ($lefty2[1] -ilt $righty2[1])
+    {
+        return -1
+    }
+    elseif ($lefty2[1] -igt $righty2[1])
+    {
+        return 1
+    }
+    else
+    {
+        return 0
+    }
+}
+
+function Get-TAEFPackageVersion
+{
+    $root = Get-ProjectRoot
+    $filename = Join-Path $root 'eng'
+    $filename = Join-Path $filename 'Version.Dependencies.xml'
+    $xml = [xml](Get-Content $filename -EA:Stop)
+    $taef = $xml.SelectSingleNode("/Dependencies/Dependency[@Name='Microsoft.Taef']")
+    return $taef.Version
+}
+
+function Get-VSWhereOffline
+{
+    # Absolute Path = "%ProgramFiles(x86)%\Microsoft Visual Studio\Installer\vswhere.exe"
+
+    $programfilesx86 = [Environment]::GetFolderPath([Environment+SpecialFolder]::ProgramFilesX86)
+    $path = Join-Path $programfilesx86 "Microsoft Visual Studio\Installer\vswhere.exe"
+    if (-not(Test-Path -Path $path -PathType Leaf))
+    {
+        return $null
+    }
+    $path
```

Comment:
Get-TAEFPackageVersion above uses "return $taef.Version", so do "return $path" here also for consistency?